README.md

```diff
@@ -1,12 +1,103 @@
-# Gluon Maps
+[![Gluon](.github/assets/gluon_logo.svg)](https://gluonhq.com)
 
 [![Build](https://github.com/gluonhq/maps/actions/workflows/build.yml/badge.svg)](https://github.com/gluonhq/maps/actions/workflows/build.yml)
 [![Maven Central](https://img.shields.io/maven-central/v/com.gluonhq/maps)](https://search.maven.org/#search|ga|1|com.gluonhq.maps)
 [![License](https://img.shields.io/github/license/gluonhq/maps)](https://opensource.org/licenses/GPL-3.0)
+[![javadoc](https://javadoc.io/badge2/com.gluonhq/maps/javadoc.svg)](https://javadoc.io/doc/com.gluonhq/maps)
+
+# Gluon Maps
 
-Gluon Maps provides an easy way to integrate [OpenStreetMaps](https://www.openstreetmap.org) into a Java application.
+Gluon Maps provides an easy way to integrate [OpenStreetMaps](https://www.openstreetmap.org) into a JavaFX application.
 It is blazing fast and offers layer overlays, multiple tile-sets, and much more.
 
-A [sample](https://github.com/gluonhq/maps/tree/master/samples/mobile) is provided in the repository to get started with the library.
+## API Overview
+
+### MapPoint
+Represents a set of latitude and longitude values.
+
+```
+MapPoint point = new MapPoint(latitude, longitude);
+```
+
+### MapLayer
+Shows the map tiles along with any user defined nodes.
+This class can be extended by overloading the `layoutLayer` method.
+
```

Comment:
Add some explanation for the code snippet, like:

For instance, the following code snippet shows how to create a custom layer that displays red circles at the given map points.